gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/ManifestBasedDataset.java

```diff
@@ -132,7 +132,8 @@ public Iterator<FileSet<CopyEntity>> getFileSetIterator(FileSystem targetFs, Cop
             copyableFile.setFsDatasets(srcFs, targetFs);
             copyEntities.add(copyableFile);
 
-            Path fromPath = srcFs.getFileStatus(fileToCopy).isDirectory() ? fileToCopy : fileToCopy.getParent();
+            // Always grab the parent since the above permission setting should be setting the permission for a folder itself
+            Path fromPath = fileToCopy.getParent();
```

Comment:
is it ever possible that `fileToCopy` would already be `/`?  e.g. because of a desire to alter `/`'s perms.

if so, will we safely handle the `null` parent?  suggest a test...

Reply:
I think `/`'s parent is `/`. need to check

Reply:
It's null, I handled it in the tests